pkg/processor/metricfrequencyprocessor/config.go

```diff
@@ -0,0 +1,79 @@
+package metricfrequencyprocessor
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+// Config defines configuration for ProcessMetrics.
+type Config struct {
+	*config.ProcessorSettings `mapstructure:"-"`
+
+	// MinPointAccumulationTime defines warm up time for the processor.
+	// Processor will not sift data points for metrics
+	// which do not have data points in processor's cache older than MinPointAccumulationTime.
+	MinPointAccumulationTime time.Duration `mapstructure:"min_point_accumulation_time"`
+
+	// ConstantMetricsReportFrequency defines minimum time between reports of a constant metric.
+	ConstantMetricsReportFrequency time.Duration `mapstructure:"constant_metrics_report_frequency"`
+
+	// LowInfoMetricsReportFrequency defines minimum time between reports of a low info metric.
+	LowInfoMetricsReportFrequency time.Duration `mapstructure:"low_info_metrics_report_frequency"`
+
+	// MaxReportFrequency defines minimum time between reports of any metric.
+	MaxReportFrequency time.Duration `mapstructure:"max_report_frequency_seconds"`
+
+	// IqrAnomalyCoef defines relative deviation from interquartile range which constitutes an anomaly.
+	// I.e. value v such that v > Iqr * IqrAnomalyCoef is considered an anomaly.
+	IqrAnomalyCoef float64 `mapstructure:"iqr_anomaly_coefficient"`
+
+	// VariationIqrThresholdCoef variation to iqr quotient above which a metric is no longer considered low info.
+	// Variation means sum of absolute values of differences between consecutive data points.
+	// I.e. if current variation v of a metric satisfies v / Iqr > VariationIqrThresholdCoef
+	// then the metric is not considered low info.
+	VariationIqrThresholdCoef float64 `mapstructure:"variation_iqr_threshold_coefficient"`
```

Comment:
It only occurred to me now:

I believe you can use

```
`mapstructure:"<property>,squash"`
```

inside the `sieveConfig` `struct` and then embed it here. Then you'll expose its field into the `Config` without the code duplication. Similarly we can simplify the properties related to `cacheConfig`.

Reply:
https://pkg.go.dev/github.com/mitchellh/mapstructure?utm_source=godoc#hdr-Embedded_Structs_and_Squashing

Reply:
squashed

Also, I've added a test for this config now.